### ISSUE TITLE
module-import-not-at-top-of-file (E402) is not reported in functions

### ISSUE THREAD

**Issue by stmlange** [astral-sh/ruff#18661]
### Summary

Hello Ruff-Team,
thanks for an amazing tool! I'm not sure if this is intentional, but the E402 rule does not seem to fire when the import is used in functions.

Consider for example:

```python
"One string"
"Two string"

import sys


class FooBar:
    def __init__(self, fpath):
        self.fpath = fpath

    def is_valid(self) -> bool:
        from pathlib import Path
        return Path(self.fpath).exists()

    @staticmethod
    def get_sep() -> None:
        from os import sep
        print(sep)
        print(sys.path)

FooBar("missing.txt")
```

or
```
def another_example():
    import platform
    print(platform.system())

another_example()
```

A `ruff check` only complains about the sys import but not about the imports that are defined in the function(s).

I encountered this issue because someone of my Team accidently imported an external dependency that was not encoded in the pyproject.toml effectively causing a `ModuleNotFoundError: No module named xyz` during runtime.

Thank you for your time and consideration.

### Version

ruff 0.11.12 (aee3af0f7 2025-05-29)

**Comment by ntBre** [comment]
I believe this is intentional and E402 only applies to module-level/global imports. You might also want to activate [import-outside-top-level (PLC0415)](https://docs.astral.sh/ruff/rules/import-outside-top-level/#import-outside-top-level-plc0415) to flag function-level imports.

And thanks for the kind words!

**Comment by stmlange** [comment]
Ahh sorry my bad. Thanks for pointing me to the right rule and your ultra fast response :-)

**Comment by ntBre** [comment]
No problem, glad I could help! :)
